analysis/02_new_model_input/03.2_computing_stats.md

```diff
@@ -0,0 +1,208 @@
+# Notebook that reads in typhoon data and computes 6 hr and 24 hr max in mm/hr
+
+```python
+%load_ext jupyter_black
+
+import pandas as pd
+import os
+from pathlib import Path
+import math
+import datetime
+
+# input
+input_dir = (
+    Path(os.getenv("STORM_DATA_DIR"))
+    / "analysis/02_new_model_input/03_rainfall/input"
+)
+
+# outputs
+processed_output_dir = os.path.join(
+    input_dir, "gpm_data/rainfall_data/output_hhr_processed/"
+)
+
+output_dir = os.path.join(
+    Path(os.getenv("STORM_DATA_DIR"))
+    / "analysis/02_new_model_input/03_rainfall/output"
+)
+typhoon_metadata = pd.read_csv(input_dir / "metadata_typhoons.csv")
+```
+
+```python
+# loading csv files
+typ_list = [
+    x for x in os.listdir(processed_output_dir) if x.endswith("_gridstats.csv")
+]
+# typ_file = pd.read_csv(processed_output_dir + typ_list[0])
+# typ_file
+```
+
+This extracts the 6 hr and 24 hr max for each hour for both the mean and max columns.
+
+```python
+# To make sure the dates can be converted to date type
+for i in range(len(typhoon_metadata)):
+    typhoon_metadata["startdate"][i] = typhoon_metadata["startdate"][
+        i
+    ].replace("/", "-")
+    typhoon_metadata["enddate"][i] = typhoon_metadata["enddate"][i].replace(
+        "/", "-"
+    )
+    typhoon_metadata["landfalldate"][i] = typhoon_metadata["landfalldate"][
+        i
+    ].replace("/", "-")
+
+typhoon_metadata["landfall_date_time"] = (
+    typhoon_metadata["landfalldate"] + "-" + typhoon_metadata["landfall_time"]
+)
+
+typhoons = list(typhoon_metadata["typhoon"].values)
+#%% Processing the data into an excel sheet
+time_frame = 48  # in half hours
+time_frame_6 = 12  # in half hours
+mov_window = 12  # in half hours
+before_landfall_h = 72  # how many hours before landfall to include
+after_landfall_h = 72  # how many hours before landfall to include
+
+num_intervals = (
+    math.floor((2 * 144 - time_frame) / mov_window) + 1
+)  # 144 to include 72hr before and after
```

Comment:
Better then to put 2 * 2 * 72.  But anyway let's see if you need all this stuff after switching to `pandas.rolling`.